pytroll_collectors/segments.py

```diff
@@ -76,6 +76,30 @@ def __init__(self, config):
         self.logger = logging.getLogger("segment_gatherer")
         self._loop = False
 
+        # Convert check time into int minutes variables
+        for key in self._patterns:
+            if "hour_pattern" in self._patterns[key]:
+                checkTime = self._patterns[key]["hour_pattern"]
+                # Ex. checkTime =  "06:00 18:00 00:30"
+                startTime, endTime, deltaTime = checkTime.split(' ')
+                if startTime != '' and endTime != '' and deltaTime != '':
```

Comment:
the suggestion above would allow you to get the values directly instead of having to split them. Also, could you make start and end times optional ? (defaulting to 00:00 and 23:59 respectively)